pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +319,419 @@ func TestMemberService(t *testing.T) {
 		})
 	}
 }
+
+func TestStatefulSetForRack(t *testing.T) {
+	basicRack := scyllav1.RackSpec{
+		Name: "rack",
+		Storage: scyllav1.StorageSpec{
+			Capacity: "1Gi",
+		},
+	}
+	basicSC := func() *scyllav1.ScyllaCluster {
+		sc := &scyllav1.ScyllaCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "basic",
+				UID:  "the-uid",
+			},
+			Spec: scyllav1.ScyllaClusterSpec{
+				Datacenter: scyllav1.DatacenterSpec{
+					Name: "dc",
+					Racks: []scyllav1.RackSpec{
+						basicRack,
+					},
+				},
+			},
+			Status: scyllav1.ScyllaClusterStatus{
+				Racks: map[string]scyllav1.RackStatus{},
+			},
+		}
+		return sc
+	}
+	basicSTSLabels := func() map[string]string {
+		return map[string]string{
+			"app":                          "scylla",
+			"app.kubernetes.io/managed-by": "scylla-operator",
+			"app.kubernetes.io/name":       "scylla",
+			"scylla/cluster":               "basic",
+			"scylla/datacenter":            "dc",
+			"scylla/rack":                  "rack",
+		}
+	}
+
+	basicSTSLabelsWithVersion := func() map[string]string {
+		m := basicSTSLabels()
+		m["scylla/scylla-version"] = ""
+		return m
+	}
+
+	basicImagePullSecrets := func() []corev1.LocalObjectReference {
+		imagePullSecrets := []corev1.LocalObjectReference{
+			{
+				Name: "basic-secrets",
+			},
+		}
+		return imagePullSecrets
+	}
+
+	newSts := func() *appsv1.StatefulSet {
```

Comment:
nit: `newExpectedStatefulSet`
would it be better to inline the values to the changes are explicit and reviewable?

Reply:
done